modules/integration/src/test/scala/scala/cli/integration/ExportCommonTestDefinitions.scala

```diff
@@ -21,6 +21,7 @@ trait ExportCommonTestDefinitions { _: ScalaCliSuite & TestScalaVersionArgs =>
   protected val prepareTestInputs: TestInputs => TestInputs = identity
 
   protected val outputDir: os.RelPath = os.rel / "output-project"
+  def majorScalaVersion: String
```

Comment:
calling it `majorScalaVersion` feels confusing... by `major` we usually mean the major version component (so the first digit of the version, as opposed to minor/patch).
I'd also not cover it by inheritance, and rather test the versions we want to check in a for loop.
for example, in the current state of the PR, we don't check exporting `3.nightly`.